packages/docusaurus-theme-classic/src/theme/DocRoot/Layout/Sidebar/styles.module.css

```diff
@@ -29,4 +29,11 @@
     width: var(--doc-sidebar-hidden-width);
     cursor: pointer;
   }
+
+  .sidebarViewport {
+    top: 0;
+    position: sticky;
+    height: 100%;
+    max-height: 100vh;
+  }
```

Comment:
`.sidebarViewport` is a sticky element that's clamped to screen height. Its width will shrink on collapse.
`.sidebarViewport`, `.sidebar`, and `.expandButton` elements will have the same height (on collapse & expand).